### PR TITLE
Implement base32 & unify base64

### PR DESCRIPTION
**Pull request by knight42**

### CODE REVIEW

File: tests/test_factor.rs [hunk 1]
```diff
@@ -8,9 +8,14 @@
...
-use rand::distributions::{IndependentSample, Range};
-use sieve::Sieve;
+
+#[path="../src/factor/sieve.rs"]
+mod sieve;
+use self::sieve::Sieve;
+
+extern crate rand;
+use self::rand::{weak_rng, Rng};
+use self::rand::distributions::{IndependentSample, Range};
```
Comment (nathanross) [inline tests/test_factor.rs]: @knight42 why are these changes in this PR?

File: tests/test_mktemp.rs [hunk 2]
```diff
@@ -1,5 +1,6 @@
 use common::util::*;
-use tempdir::TempDir;
+extern crate tempdir;
+use self::tempdir::TempDir;
```
Comment (nathanross) [inline tests/test_mktemp.rs]: @knight42 see above

File: tests/test_split.rs [hunk 3]
```diff
@@ -4,8 +4,8 @@ extern crate regex;
 use std::fs::{File, read_dir};
 use std::io::{Read, Write};
 use std::path::Path;
-use rand::{Rng, thread_rng};
-use regex::Regex;
+use self::rand::{Rng, thread_rng};
+use self::regex::Regex;
```
Comment (nathanross) [inline tests/test_split.rs]: @knight42 see above

File: tests/tests.rs [hunk 4]
```diff
@@ -1,19 +1,6 @@
...
-extern crate libc;
-extern crate rand;
-extern crate regex;
-extern crate tempdir;
-extern crate time;
-
-#[cfg(windows)] extern crate kernel32;
-#[cfg(windows)] extern crate winapi;
-
 #[macro_use]
```
Comment (nathanross) [inline tests/tests.rs]: a very great change, but see above - re pr

File: tests/test_base64.rs [hunk 5]
```diff
@@ -32,8 +32,7 @@ fn test_garbage() {
     new_ucmd()
         .arg("-d")
         .pipe_in(input)
-        .fails()
-        .stderr_only("base64: error: invalid character (Invalid character '0' at position 20)\n");
```
Comment (nathanross) [inline tests/test_base64.rs]: If these are GNU error messages, good job on removing them, if not please leave them in.

**Comment by nathanross** [comment]
Love it. Please break into two different PRs and exclude the changes in test_base64.rs which reduce coverage.

**Comment by knight42** [comment]
@nathanross The CI breakage is related to a [bug](https://internals.rust-lang.org/t/cargo-regression-regex-crate-broken-on-2016-08-05-nightly/3819) in nightly cargo, I can do nothing at present.

**Comment by knight42** [comment]
@nathanross Pass all tests now. Any other problems?

**Comment by nathanross** [comment]
awesome

**Pull request merged**